code/src/components/Task.js

```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import { useDispatch } from 'react-redux'
+import moment from 'moment'
+
+import { tasklist } from '../reducers/tasklist'
+import { Checkbox } from './Checkbox'
+import { TaskContainer, TaskText, Button, Time } from './styled/tasks'
+
+export const Task = ({ id, text, complete }) => {
+  const dispatch = useDispatch()
+
+  const handleCheckToggle = () => {
+    dispatch(tasklist.actions.toggleCheck(id))
+  }
+
+  const handleRemove = () => {
+    dispatch(tasklist.actions.removeTodo(id))
+  }
+
+  return (
+    <>
+      <TaskContainer>      
+          <Checkbox 
+            isChecked={complete}
+            onChangeHandler={handleCheckToggle}
+          />
+          <TaskText>{text}</TaskText>
+        <Button onClick={handleRemove}><span role='img' aria-label='Delete'> 🗑️</span></Button>
+      </TaskContainer>
+      <Time>
+        <p>Added: {moment().calendar()}</p>
+      </Time>
```

Comment:
Really awesome that you implemented the task date and time when it was added!

Reply:
As an add on - maybe you can hard code the times for the tasks in your initial state and add a field to your action that adds todos as well (to add a date when a task is created), because right now all the task gets the same date as the latest submitted one :)